lib/rules/prefer-lookaround.ts

```diff
@@ -0,0 +1,659 @@
+ */
+class ReplaceReferencesList {
+    private readonly list: ReplaceReferences[]
+
+    /** Reference name at the starting position. */
+    public readonly startRefName?: string | number
+
+    /** Reference name at the ending position. */
+    public readonly endRefName?: string | number
+
+    /** All reference names except at the starting position. */
+    public readonly otherThanStartRefNames: Set<string | number>
+
+    /** All reference names except at the starting position. */
+    public readonly otherThanEndRefNames: Set<string | number>
+
+    public constructor(list: ReplaceReferences[]) {
+        this.list = list
+        this.startRefName = list[0].startRef?.ref
+        this.endRefName = list[0].endRef?.ref
+
+        const otherThanStartRefNames = new Set<string | number>()
+        const otherThanEndRefNames = new Set<string | number>()
+        for (const { startRef, endRef, allRefs } of this.list) {
+            for (const ref of allRefs) {
+                if (ref !== startRef) {
+                    otherThanStartRefNames.add(ref.ref)
+                }
+                if (ref !== endRef) {
+                    otherThanEndRefNames.add(ref.ref)
+                }
+            }
+        }
+        this.otherThanStartRefNames = otherThanStartRefNames
+        this.otherThanEndRefNames = otherThanEndRefNames
+    }
+
+    public *[Symbol.iterator](): Iterator<ReplaceReferences> {
+        yield* this.list
+    }
+}
+
+const enum SideEffect {
+    startRef,
+    endRef,
+}
+
+/**
+ * Gets the type of side effect when replacing the capture group for the given element.
+ *
+ * There are no side effects if the following conditions are met:
+ *
+ * - Some elements other than the start capturing group have disjoints to the start capturing group.
+ * - The last element and the start consume character have disjoint.
+ */
+function getSideEffectsWhenReplacingCapturingGroup(
+    start: CapturingGroup | undefined,
+    others: Element[],
+    end: CapturingGroup | undefined,
+    { flags }: RegExpContext,
+): Set<SideEffect> {
+    const result = new Set<SideEffect>()
+    if (start) {
+        if (!hasDisjoint(start, end ? [...others, end] : others)) {
+            result.add(SideEffect.startRef)
+        }
+    }
+
+    const last = end || others[others.length - 1]
+    const first = start || others[0]
+    const lastChars = getPossiblyConsumedChar(last, flags)
+    const firstChar = getFirstConsumedChar(first, "ltr", flags)
+    if (!firstChar.char.isDisjointWith(lastChars.char)) {
+        result.add(SideEffect.startRef)
+        result.add(SideEffect.endRef)
+    }
+    return result
+
+    /** Checks whether the given target element has disjoint in elements.  */
+    function hasDisjoint(target: Element, elements: Element[]) {
+        const chars = getPossiblyConsumedChar(target, flags)
+        for (const element of elements) {
+            if (isConstantLength(element)) {
```

Comment:
Sorry, I made a mistake. We also have to make sure that the element consumes at least one character.

Assertions have a constant length of 0 but don't consume any characters, `getPossiblyConsumedChar(assertion)` is the empty set and the empty set is disjoint with everything.

Reply:
Thank you for letting me know! I will fix it.